src/proto/nc_redis.c

```diff
@@ -2028,7 +2074,13 @@ redis_parse_rsp(struct msg *r)
         case SW_BULK_ARG_LF:
             switch (ch) {
             case LF:
-                goto done;
+                if (r->depth > 0) {
+                    if (update_and_check_done(r) == true) {
```

Comment:
even though this works, stylistically returning int and comparing boolean is not good.